### DISCUSSION TITLE
LibreChat API

### DISCUSSION THREAD

**Discussion by tip-dteller** [danny-avila/LibreChat#4679] [Q&A]
Hi,
Ive noticed there's a "librechat-api" image as part of the release, i suppose its the same without the UI part.
Is there any documentation on how to use it? 
The use-case i have is i'd like to use Librechat in CI for example, and feed it data while using API.
so what matters is - how can i authenticate a login, use assistants, use normal chat and upload files and use them in chat/assistant, all via API.

Thanks.

**Comment by danny-avila** [comment]
The LibreChat API is the backend that runs and serves the application.

**Reply by d-teller** [reply]
Which further strengthens my question.
Is there a doc detailing the endpoints? Like a swaggerUI?

**Reply by danny-avila** [reply]
No that’s not its intended usage

**Reply by d-teller** [reply]
I understand, thank you Danny, i really appreciate Librechat and the effort you've put into it!


For the sake of discussion, in your opinion, is it even possible given the current code base or a significant refactoring effort would be required?
